src/shared/Microsoft.Git.CredentialManager.Tests/InputArgumentsTests.cs

```diff
@@ -62,5 +62,78 @@ public void InputArguments_OtherArguments()
             Assert.Equal("bar", inputArgs["foo"]);
             Assert.Equal("bar", inputArgs.GetArgumentOrDefault("foo"));
         }
+
+        [Fact]
+        public void InputArguments_GetRemoteUri_NoAuthority_ReturnsNull()
+        {
+            var dict = new Dictionary<string, string>();
+
+            var inputArgs = new InputArguments(dict);
+
+            Uri actualUri = inputArgs.GetRemoteUri();
+
+            Assert.Null(actualUri);
+        }
+
+        [Fact]
+        public void InputArguments_GetRemoteUri_Authority_ReturnsUriWithAuthority()
+        {
+            var expectedUri = new Uri("https://example.com/");
+
+            var dict = new Dictionary<string, string>
+            {
+                ["protocol"] = "https",
+                ["host"]     = "example.com"
+            };
+
+            var inputArgs = new InputArguments(dict);
+
+            Uri actualUri = inputArgs.GetRemoteUri();
+
+            Assert.NotNull(actualUri);
```

Comment:
Curious here, why do a NotNull then an Equal check.  Would the equal check be sufficient?

Reply:
For me it makes the error reporting slightly nicer in case of failure. :)